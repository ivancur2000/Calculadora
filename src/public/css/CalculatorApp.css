html{
    font-family: -apple-system,sans-serif;
}
body{
    background-color: rgb(71, 83, 100);
    margin: 50px;
}
.calculator__content{
    background-color: rgb(94, 108, 128);
    border: 4px solid rgb(197, 210, 224);
    border-radius: 20px;
    padding: 50px;
    width: 500px;
}
.btn{
    width: 45px
}
button{
    margin: 5px;
    padding: 5px;
}