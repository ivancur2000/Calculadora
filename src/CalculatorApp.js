import React from 'react';
import { Keyboard } from "./components/Keyboard";
import { InputDisplay } from "./components/InputDisplay";
import { useCalculate } from './hooks/useCalculate';
import './public/css/CalculatorApp.css';

export const CalculatorApp = ()=>{

    const { 
        inputValue,
        handleOnClickNumber,
        handleOnClickOperation,
        handleOnReset,
        handleOnResolve 
    } = useCalculate();

    return(
        <div className='calculator__content'>
            <h1>Calculator App!!!!</h1>
            <InputDisplay value={inputValue} />
            <Keyboard 
                handleOnClickNumber={handleOnClickNumber} 
                handleOnClickOperation={handleOnClickOperation}
                handleOnReset={handleOnReset}
                handleOnResolve={handleOnResolve}
            />
        </div>
    );
}