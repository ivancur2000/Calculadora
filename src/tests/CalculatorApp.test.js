import { shallow } from "enzyme";
import { CalculatorApp } from "../CalculatorApp";

describe('Pruebas en <CalculatorApp />',  ()=> {

    const wrapper = shallow(<CalculatorApp />);
    test('El componente debe renderizarse correctamente', ()=>{
        expect(wrapper).toMatchSnapshot();
    });
    test('El componente <Keyboard /> debe existir', () => {
        const keyboard = wrapper.find('Keyboard');
        expect(keyboard.exists()).toBeTruthy();
    });
    test('El componente <InputDisplay /> debe existir', () => {
        const inputDisplay = wrapper.find('InputDisplay');
        expect(inputDisplay.exists()).toBeTruthy();
    });
});