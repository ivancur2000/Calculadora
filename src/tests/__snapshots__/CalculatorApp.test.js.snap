// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CalculatorApp /> El componente debe renderizarse correctamente 1`] = `
<div
  className="calculator__content"
>
  <h1>
    Calculator App!!!!
  </h1>
  <InputDisplay
    value={Array []}
  />
  <Keyboard
    handleOnClickNumber={[Function]}
    handleOnClickOperation={[Function]}
    handleOnReset={[Function]}
    handleOnResolve={[Function]}
  />
</div>
`;
